packages/expo-av/ios/EXAV/Video/EXVideoView.m

```diff
@@ -129,15 +139,19 @@ - (void)_removePlayer
     _requestedFullscreenChange = NO;
   }
 
-  if (_data) {
-    [_data pauseImmediately];
-    [_data setStatusUpdateCallback:nil];
-    [_exAV demoteAudioSessionIfPossible];
-    [self _removeFullscreenPlayerViewController];
-    [self _removePlayerLayer];
-    [self _removePlayerViewController];
-    _data = nil;
-  }
+  // Any ViewController/layer updates need to be
+  // executed on the main UI thread.
+  __weak EXVideoView *weakSelf = self;
```

Comment:
FYI, we have macros for handling those nasty `__weak self` references.
Instead of `__weak VariableType *var = variable` use `UM_WEAKIFY(variable)`
and instead of `__strong ...` use `UM_STRONGIFY` or even `UM_ENSURE_STRONGIFY` (the latter comes with auto-checking for nullability).